Trappist/src/Promact.Trappist.Core/Controllers/TestController.cs

```diff
@@ -11,30 +10,38 @@ namespace Promact.Trappist.Core.Controllers
     public class TestsController : Controller
     {
         private readonly ITestsRepository _testRepository;
-        private readonly IStringConstants _stringConstant;
-  
-        public TestsController(ITestsRepository testRepository, IStringConstants stringConstant)
+        public TestsController(ITestsRepository testRepository)
         {
             _testRepository = testRepository;
-          _stringConstant = stringConstant;
-        }    
+        }
+        /// <summary>
+        /// this method is to verify unique name of a test
+        /// </summary>
+        /// <param name="testName">name of the test</param>
+        /// <returns>boolean</returns>
+        [HttpGet("{testName}")]
+        public async Task<IActionResult> IsUniqueTestName([FromRoute] string testName)
+        {
+            // verifying the test name is unique or not
+            bool isExist = await _testRepository.IsTestNameUniqueAsync(testName);
+            return Ok(isExist);          
+        }
         /// <summary>
         /// this method is used to add a new test 
         /// </summary>
-        /// <param name="test">object of the Test</param>
-        /// <returns>string</returns>
+        /// <param name="test">object of the Test</param>   
+        /// <returns>created test</returns>
         [HttpPost]
-        public string CreateTest([FromBody] Test test)
+        public async Task<IActionResult> CreateTest([FromBody] Test test)
         {
-            if (_testRepository.UniqueTestName(test)) // verifying the test name is unique or not
+            if (ModelState.IsValid)
             {
-                _testRepository.CreateTest(test);
-                return _stringConstant.Success;              
+                _testRepository.RandomLinkString(test, 10);
```

Comment:
Why would you create separate public method for generating random string. thats wrong.

RandomLinkString name doesnt indicate anything (other dev cant know what this method does until he/she sees the code).

What if other developer wants random string method in his repository also? will he create same method as u? Pls change it so others can also use